services/crosschainconnector/ethereum/adapter/simulator.go

```diff
@@ -0,0 +1,45 @@
+package adapter
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/crypto"
+	"math/big"
+)
+
+type EthereumSimulator struct {
+	auth      *bind.TransactOpts
+	simClient bind.ContractBackend
+}
+
+func NewEthereumSimulatorConnector() EthereumConnection {
+	return &EthereumSimulator{}
+}
+
+func (es *EthereumSimulator) Dial(endpoint string) error {
+	// Generate a new random account and a funded simulator
+	key, err := crypto.GenerateKey()
+	if err != nil {
+		return err
+	}
+	es.auth = bind.NewKeyedTransactor(key)
+
+	genesisAllocation := map[common.Address]core.GenesisAccount{
+		es.auth.From: {Balance: big.NewInt(10000000000)},
+	}
+
+	sim := backends.NewSimulatedBackend(genesisAllocation, 900000000000)
+	es.simClient = sim
```

Comment:
This function is running potentially from multiple goroutines, it doesn't look very thread-safe

Reply:
you are right, but it is supposed to be test only code, i'll protect it because it makes no difference, this does makes me think i should go over the entire connection flow because i think i will mess up the real node client creation because of a rece condition, will fix that as well